### DISCUSSION TITLE
Hot Chocolate with Azure Functions net-iso Http Trigger functions?

### DISCUSSION THREAD

**Discussion by kosperera** [ChilliCream/hotchocolate#5351] [Q&A] [answered]
I am looking for recommendations, docs, or code examples on how to use Hot Chocolate with Azure Functions net-iso apps, say, similar to ASP.NET Web API but written using Azure Functions net-iso apps with HTTP triggers.

**Comment by cajuncoding** [comment, marked as answer]
What exactly do you mean by "net-iso"? Are you referring to running Azure Function in the [Isolated Process](https://docs.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide) model?

If so, then that is not currently officially supported by Hot Chocolate out-of-the-box. . . _yet_. But it's on the way soon! 

The work has been committed into v13 future release, and is slated to also be included in the new upcoming release for v12.13.

Until then your best (and fully working option for v12) has been the [GraphQL.AzureFunctionsProxy](https://github.com/cajuncoding/GraphQL.AzureFunctionsProxy) library -- which will be deprecated once the official support is released for Isolated Process model . . . because the solution from that project has been merged with the official code and improved a little.

**Reply by kosperera** [reply]
Thanks, @cajuncoding. I came across the library that you mentioned. I wouldn't want to use that since it is pulling [ASP.NET Core dependencies](https://github.com/cajuncoding/GraphQL.AzureFunctionsProxy/blob/606799f0e0f94518cad559b92bcf409564344706/GraphQL.AzureFunctionsProxy.IsolatedProcess/GraphQL.AzureFunctionsProxy.IsolatedProcess.csproj#L28-L35) into Function Workers. I was looking for a [vanilla Azure Functions .NET 6+ Isolated Process](https://docs.microsoft.com/en-us/azure/azure-functions/create-first-function-cli-csharp?tabs=azure-cli%2Cisolated-process) code sample to get started.

**Reply by cajuncoding** [reply]
Well the dependency is absolutely minimal but HttpContext is required for HC core execution engine… so FYI, you will have the same dependency even with the new official support coming out (now merged into v13 preview).

**Reply by kosperera** [reply]
@cajuncoding You are absolutely right. I just checked the dependencies and it is pulling [HC ASP.NET](https://github.com/ChilliCream/hotchocolate/blob/5991f99f6d4c43242197d8417beec7212bfc9775/src/HotChocolate/AzureFunctions/src/HotChocolate.AzureFunctions/HotChocolate.AzureFunctions.csproj#L19) which [has the ASP.NET SDK](https://github.com/ChilliCream/hotchocolate/blob/5991f99f6d4c43242197d8417beec7212bfc9775/src/HotChocolate/AspNetCore/src/AspNetCore/HotChocolate.AspNetCore.csproj#L24).

Do you happen to know if there are any plans to remove ASP.NET SDK dependency from the HC Functions (Isolated) package?

**Reply by cajuncoding** [reply]
No, since it was finally just added there’s no current plans to remove that dependency. As I stated above the `HttpContext` is critical/required by HC core execution process so significant changes would be needed to change.

It’s far easier and efficient (ie. to get support added/included) to adapt the Isolated Process `HttpRequestData` to what HC already consumes…and this approach is the same as that of the AzureFunctionsProxy library because we collaborated to merge and then optimize it for official inclusion.

TBH I’m not sure what the concern is as the dependency will have no impact to performance.

**Comment by bo55vxr** [comment]
Hi @kosalanuwan 

I am using GraphQL in the same way you have indicated (Azure Function / .Net Isolated) .... Is there a specific question you have right now?

The documentation on the CC website assumes that everyone uses WebAPI which does make things a little..... challanging ;)

**Reply by kosperera** [reply]
@bo55vxr Cool. Is it correct to assume you are not using the library that @cajuncoding mentioned in his [answer](https://github.com/ChilliCream/hotchocolate/discussions/5351#discussioncomment-3491856)? 
If so, do you happen to have a code sample in public for me to refer to how it is been done?

**Comment by cajuncoding** [comment]
Az Func with Isolated process is now officially supported in v13 preview, and the new video from Michael demonstrates it using the latest v13 preview:

https://youtu.be/TTbyHeGDSIk

Hopefully it’ll be back ported to v12 in a soon upcoming release also… 

The new template project shows how easy it will be to use the new official implementation…and migrating from the AzureFunctionProxy library should be very straightforward and easy.
